index.template.html

```diff
@@ -5009,6 +4990,38 @@ <h2>Required</h2>
 
 </section>
 
+    <section id="thing-model-composition" class="normative">
+        <h3>Composition</h3>
+<p>
+    In some applications, it is beneficial to reuse existing <a>Thing Model</a> definitions and compose them to a new IoT system. 
+    An example can be that a Smart Ventilator <a>Thing Model</a> consist of two <a>Thing Model</a> definitions such as 
+    a Ventilation <a>Thing Model</a> that provides <code>on/off</code> and <code>adjustRpm</code> capabilities, and an LED <a>Thing Model</a> that 
+    provides <code>dimmable</code> and <code>RGB</code> capabilities. 
+</p>
+
+
+<pre class="example" title="Thing Model witht the compose feature.">
+    {
+        "@context": "http://www.w3.org/ns/td",
+        "@type": "tm:ThingModel",
+        "title": "Smart Ventilator Thing Model",
+        "version" : { "model" : "1.0.0" },
+        "links": [
+          {
+            "rel": "subthing",
+            "href": "./ventilation.tm.jsonld",
+            "instanceName": "ventilation"
+          },
+          {
+            "rel": "subthing",
+            "href": "./led.tm.jsonld",
+            "instanceName": "led"
+          }
+        ]
```

Comment:
@sebastiankb Thank you for adding a compose mechanism! I think this already looks promising. Some initial thoughts:

I was wondering if an approach similar to the one chosen by SDF (using objects and things in the TM itself) could be an alternative (or addition?) as well. Using a similar structure could make it easier to reuse models from SDF and therefore also from other ecosystems.

Also, I guess you would have to "flatten" the `subthing` links contained in a TM when converting such a TM to a TD, right? In this case, would the `instanceName` be used as a prefix of all affordances? And would top-level fields like the `@context` be merged into one? Or would the TM that is used for composition overwrite the values from the referenced TMs?

Reply:
Thanks for the feedback. Definitely, I would be open for evaluation. Can you share an example how you can refer to existing models without defining again within the object container?  Would this example work:

```
    {
        "@context": "http://www.w3.org/ns/td",
        "@type": "tm:ThingModel",
        "title": "Smart Ventilator Thing Model",
        "version" : { "model" : "1.0.0" },
        "tmObject": {
           "ventilation" :  { "tmRef": "./ventilation.tm.jsonld"}
            "led": {"tmRef": "./led.tm.jsonld"}
          }
      }
```

Reply:
Very nice :) However, maybe we need to discuss if in this case the `tm:ref` should refer to the whole Thing Model or rather to another object within the TM (for example using `"ventilation" :  { "tmRef": "./ventilation.tm.jsonld#tmObject/ventilation"}`. I think both approaches could make sense. Having multiple `@context` fields in the case of referencing a complete TM wouldn't be a problem, would it?

Reply:
In my understanding, each object entry is an own Thing. Usually, for each Thing should be an own TM/TD. The original idea is to provide a way to describe that your system is based on a set of subsystems which each of them would have its own TD/TM definition. If TDs are generated this hierarchy can be provided by the links container again:

```json
"links": [{
     "rel": "item",
     "href": "path/to/ventilation.td.json"
 },
 {
     "rel": "item",
     "href": "path/to/led.td.json"
 },
```